src/containers/stage.jsx

```diff
@@ -32,10 +33,12 @@ class Stage extends React.Component {
         this.attachRectEvents();
         this.attachMouseEvents(this.canvas);
         this.updateRect();
-        this.renderer = new Renderer(this.canvas);
-        this.props.vm.attachRenderer(this.renderer);
         this.audioEngine = new AudioEngine();
         this.props.vm.attachAudioEngine(this.audioEngine);
+        this.renderer = new Renderer(this.canvas);
+        this.props.vm.attachRenderer(this.renderer);
+        this.storage = new Storage();
+        this.props.vm.attachStorage(this.storage);
```

Comment:
I think my preference would be in the GUI container — the default vm prop should be an instance of the VM with the storage attached.  That way if someone wanted to pass in a vm with a different storage attached, they could.  And we would maintain the base GUI component being "batteries included".